### PR TITLE
rgw: update bucket cors in secondary zonegroup should forward to master

### PR DESCRIPTION
**Pull request by shashalu**
#14082 is incomplete, cors should be redirect to master

Fixes: http://tracker.ceph.com/issues/16888

Signed-off-by: Shasha Lu <lu.shasha@eisoo.com>

### CODE REVIEW

**Comment by cbodley** [comment]
thanks @shashalu, could you please add a simple test for this in `src/test/rgw/rgw_multi/tests.py`? something similar to `test_bucket_acl()`, that verifies that the change is applied immediately on non-master zones without waiting for metadata sync

**Comment by shashalu** [comment]
@cbodley  add test_bucket_cors() in tests.py

File: src/test/rgw/rgw_multi/tests.py [hunk 1]
```diff
@@ -661,6 +662,14 @@ def test_bucket_acl():
         bucket.set_acl('public-read')
         assert(len(bucket.get_acl().acl.grants) == 2) # new grant on AllUsers
 
+def test_bucket_cors():
+    buckets, zone_bucket = create_bucket_per_zone_in_realm()
+    for zone, bucket in zone_bucket.items():
```
Comment (cbodley) [inline src/test/rgw/rgw_multi/tests.py]: after rebasing this on master, i'm seeing this failure:
```
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose-1.3.7-py2.7.egg/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/usr/lib/python2.7/site-packages/nose-1.3.7-py2.7.egg/nose/util.py", line 620, in newfunc
    return func(*arg, **kw)
  File "/home/cbodley/ceph/src/test/rgw/rgw_multi/tests.py", line 660, in test_bucket_cors
    for zone, bucket in zone_bucket.items():
AttributeError: 'list' object has no attribute 'items'
```
commit f5037624fd4a96e350747be3355d9ab24114189a changed `zone_bucket` from dict to list, so this line needs to be `for zone, bucket in zone_bucket:`

**Comment by cbodley** [comment]
@shashalu please rebase

**Comment by cbodley** [comment]
jenkins test this please

**Pull request merged**